### PR TITLE
refactor(WebSocketManager): passing in strategy

### PR DESCRIPTION
**Pull request by didinele**
**Please describe the changes this PR makes and why it should be merged:**
Removes `manager#setStrategy` in favor of a new `buildStrategy` manager option.

**Status and versioning classification:**
- Code changes have been tested against the Discord API, or there are no code changes
- I know how to update typings and have done so, or typings don't need updating
- This PR changes the library's interface (methods or parameters added)
- This PR includes breaking changes (methods removed or renamed, parameters moved or removed)

### CODE REVIEW

File: packages/ws/README.md [hunk 1]
```diff
@@ -107,12 +107,11 @@ const manager = new WebSocketManager({
 	intents: 0,
 	rest,
 	shardCount: 6,
+	// This will cause 3 workers to spawn, 2 shards per each
+	buildStrategy: (manager) => new WorkerShardingStrategy(manager, { shardsPerWOrker: 2 }),
```
Comment (vladfrangu) [inline packages/ws/README.md:111]:
```suggestion
	buildStrategy: (manager) => new WorkerShardingStrategy(manager, { shardsPerWorker: 2 }),
```

File: packages/ws/README.md [hunk 2]
```diff
@@ -126,14 +125,12 @@ const manager = new WebSocketManager({
 	token: process.env.DISCORD_TOKEN,
 	intents: 0,
 	rest,
+	buildStrategy: (manager) =>
+		new WorkerShardingStrategy(manager, {
+			shardsPerWOrker: 2,
```
Comment (vladfrangu) [inline packages/ws/README.md:130]:
```suggestion
			shardsPerWorker: 2,
```

File: packages/ws/src/ws/WebSocketManager.ts [hunk 3]
```diff
@@ -71,6 +70,20 @@ export interface RequiredWebSocketManagerOptions {
...
+	/**
+	 * Builds the strategy to use for sharding
+	 *
+	 * @example
+	 * ```ts
+	 * const manager = new WebSocketManager({
+	 * 	token: process.env.DISCORD_TOKEN,
+	 *  intents: 0, // for no intents
+	 *  rest,
+	 *  buildStrategy: (manager) => new WorkerShardingStrategy(manager, { shardsPerWorker: 2 }),
```
Comment (vladfrangu) [inline packages/ws/src/ws/WebSocketManager.ts:82]: Indentation please

**Review by vladfrangu — CHANGES_REQUESTED** [review]

File: packages/ws/src/ws/WebSocketManager.ts (hunk 3 above)
Comment (didinele) [reply packages/ws/src/ws/WebSocketManager.ts:82]: classic case of "it looked fine in my editor"

**Review by SpaceEEC — APPROVED** [review]

**Review by kyranet — APPROVED** [review]

**Review by vladfrangu — APPROVED** [review]

**Pull request merged**